src/v/redpanda/admin_server.h

```diff
@@ -70,6 +71,50 @@ class admin_server {
               path, std::move(req), std::move(rep));
         }
     };
+
+    enum auth_level {
+        // Unauthenticated endpoint (not a typo, 'public' is a keyword)
+        publik = 0,
+        // Requires authentication (if enabled) but not superuser status
+        user = 1,
+        // Requires authentication (if enabled) and superuser status
+        superuser = 2
+    };
+
+    /**
+     * Helper for binding handlers to routes, which also adds in
+     * authentication step and common request logging.
+     */
+    template<auth_level required_auth, bool peek_auth = false, typename F>
+    void register_route(ss::httpd::path_description const& path, F handler) {
+        path.set(
+          _server._routes,
+          [this, handler](std::unique_ptr<ss::httpd::request> req) {
+              auto auth_state = _auth.authenticate(*req);
+              if constexpr (required_auth == auth_level::superuser) {
+                  auth_state.require_superuser();
+              } else if constexpr (required_auth == auth_level::user) {
+                  auth_state.require_authenticated();
+              } else {
+                  auth_state.pass();
+              }
+
+              // Note: a request is only logged if it does not throw
+              // from authenticate().  We
+              log_request(*req, auth_state);
+
+              if constexpr (peek_auth) {
+                  return handler(std::move(req), auth_state);
+              } else {
+                  return handler(std::move(req));
+              }
+          });
+    }
+
+    void log_request(
+      ss ::httpd::request const& req,
```

Comment:
nit: i thought the whole code base was west-const (consistency helps lower cognitive load). [i like east-const but we went w/ west-const]

Reply:
I think we have ended up with a mixture (but I moved these around because you weren't the only one who noticed!).  I usually like to delegate this kind of thing to clang-format, so this sent me down a rabbit hole and... once we are on clang >= 14, there will be a `QualifierAlignment` format option for just this.